realm/realm-library/src/androidTest/java/io/realm/LinkingObjectsManagedTests.java

```diff
@@ -0,0 +1,423 @@
+    @Test
+    @Ignore
+    public void queryEndingWithBacklink() {
+        realm.beginTransaction();
+        AllJavaTypes gen1 = realm.createObject(AllJavaTypes.class, 10);
+
+        AllJavaTypes gen2A = realm.createObject(AllJavaTypes.class, 1);
+        gen2A.setFieldObject(gen1);
+
+        AllJavaTypes gen2B = realm.createObject(AllJavaTypes.class, 2);
+        gen2B.setFieldObject(gen1);
+
+        AllJavaTypes gen3 = realm.createObject(AllJavaTypes.class, 3);
+        RealmList<AllJavaTypes> parents = gen3.getFieldList();
+        parents.add(gen2A);
+        parents.add(gen2B);
+
+        realm.commitTransaction();
+
+        RealmResults<AllJavaTypes> result = realm.where(AllJavaTypes.class)
+            .isNotNull("objectParents.listParents")
+            .findAll();
+        assertEquals(2, result.size());
+        assertTrue(result.contains(gen2A));
+        assertTrue(result.contains(gen2B));
+    }
+
+    // Query on a field descriptor that has a backlink in the middle
+    @Test
+    @Ignore
+    public void queryBacklinkInMiddle() {
+        realm.beginTransaction();
+        AllJavaTypes gen1 = realm.createObject(AllJavaTypes.class, 10);
+
+        AllJavaTypes gen2A = realm.createObject(AllJavaTypes.class, 1);
+        gen2A.setFieldObject(gen1);
+
+        AllJavaTypes gen2B = realm.createObject(AllJavaTypes.class, 2);
+        gen2B.setFieldObject(gen1);
+
+        AllJavaTypes gen3 = realm.createObject(AllJavaTypes.class, 3);
+        RealmList<AllJavaTypes> parents = gen3.getFieldList();
+        parents.add(gen2A);
+        parents.add(gen2B);
+
+        realm.commitTransaction();
+
+        RealmResults<AllJavaTypes> result = realm.where(AllJavaTypes.class)
+            .lessThan("objectParents.listParents.fieldId", 4)
+            .findAll();
+        assertEquals(2, result.size());
+    }
+
+    // A newly added notification callback should be called immediately for an object
+    // that has acquired a backlink
+    // !!!FIXME What does this test?
+    @Test
+    public void notifcationSentInitially() {
+    }
+
+    // A notification callback should be called on a commit that adds a backlink
+    @Test
+    @RunTestInLooperThread
+    public void notificationOnCommit() {
+        Realm ltRealm = looperThread.realm;
+
+        ltRealm.beginTransaction();
+        AllJavaTypes gen1 = ltRealm.createObject(AllJavaTypes.class, 10);
+        ltRealm.commitTransaction();
+
+        final AtomicInteger counter = new AtomicInteger(0);
+        RealmChangeListener<AllJavaTypes> listener = new RealmChangeListener<AllJavaTypes>() {
+            @Override
+            public void onChange(AllJavaTypes object) {
+                counter.incrementAndGet();
+                looperThread.testComplete();
+            }
+        };
+        gen1.addChangeListener(listener);
+
+        ltRealm.beginTransaction();
+
+        AllJavaTypes gen2A = ltRealm.createObject(AllJavaTypes.class, 1);
```

Comment:
So that it is easier to understand what refers to what.  I suppose I could go for "child" "mom", "dad", "granddad".